### PR TITLE
Remove OverrideAuthMiddleware because it's unused

### PR DESCRIPTION
**Pull request by mrnugget**
I'm pretty sure we don't need this anymore and when in doubt I'd like to remove _overrides_ to our authentication.

The instance at `sourcegraph.sgdev.org` that's mentioned in `package.json` doesn't exist anymore and I can't find other references to the env var or header anymore.

## Test plan

- N/A

### CODE REVIEW

**Review by evict — APPROVED** [review]

**Comment by sg-e2e-regression-test-bob** [comment]
## Bundle size report 📦

| Initial size | Total size | Async size | Modules |
| --- | --- | --- | --- |
| 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0.00 kb) | 0.00% (0) |

Look at the [Statoscope report](https://console.cloud.google.com/storage/browser/_details/sourcegraph_reports/statoscope-reports/mrn/remove-override-auth/compare-report.html;tab=live_object) for a full comparison between the commits 05b401b and fd99093 or [learn more](https://docs.sourcegraph.com/dev/how-to/testing#bundlesize).


<details>
<summary>Open explanation</summary>

- `Initial size` is the size of the initial bundle (the one that is loaded when you open the page)
- `Total size` is the size of the initial bundle + all the async loaded chunks
- `Async size` is the size of all the async loaded chunks
- `Modules` is the number of modules in the initial bundle
</details>

File: cmd/frontend/internal/cli/http.go [hunk 1]
```diff
@@ -108,16 +108,12 @@ func newExternalHTTPHandler(
 	// Wrap in middleware, first line is last to run.
 	//
 	// 🚨 SECURITY: Auth middleware that must run before other auth middlewares.
-	// OverrideAuthMiddleware allows us to inject an authentication token via an
-	// environment variable, for testing. This is true only when a site-config
-	// change is explicitly made, to enable this token.
```
Comment (eseliger) [inline cmd/frontend/internal/cli/http.go:113]: what is this site config setting and can we remove it as well?

**Review by eseliger — APPROVED** [review]

File: cmd/frontend/internal/cli/http.go (hunk 1 above)
Comment (mrnugget) [reply cmd/frontend/internal/cli/http.go:113]: I honestly couldn't find anything related to that. It all seems pretty self-contained in this the files I deleted here.

**Review by kopancek — COMMENTED** [review]
I mean, the changes look good to me, but I have no idea what this target was used for. Tagged @sourcegraph/dev-experience to review as well.

**Review by kopancek — APPROVED** [review]

**Comment by mrnugget** [comment]
I also asked @valerybugakov and he said that he's never seen it in use, so I think we're pretty safe here. I also ran `main-dry-run` on the changes and everything passed.

**Pull request merged**